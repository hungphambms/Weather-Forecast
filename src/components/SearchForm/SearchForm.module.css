.searchContainer {
  padding-top: 40px;
}

.searchContainer input {
  height: 48px;
  border-color: #ececec;
  color: var(--color-dark);
  font-size: 15px;
  text-align: center;
}

.searchContainer h1 {
  font-size: 24px;
  color: var(--color-dark);
  text-transform: capitalize;
  padding: 0;
  margin: 0 0 8px;
  text-align: center;
}

.searchContainer p {
  font-size: 16px;
  color: var(--color-medium);
  text-align: center;
  padding: 0;
  margin: 0 0 24px;
}
