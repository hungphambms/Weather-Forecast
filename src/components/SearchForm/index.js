import React, { useCallback, useState } from "react";
import PropTypes from "prop-types";
import _ from "lodash";

import { Input } from "reactstrap";

import styles from "./SearchForm.module.css";

function SearchForm(props) {
  const [query, setQuery] = useState(props.query || "");
  const { onSearch } = props;

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const debounceDropDown = useCallback(
    /* istanbul ignore next */
    _.debounce((nextValue) => onSearch(nextValue), 800),
    []
  );

  function handleInputChange(value) {
    setQuery(value);
    debounceDropDown(value);
  }

  return (
    <div className={styles.searchContainer}>
      <h1>Weather Forecast</h1>
      <p>Find the most current and reliable 5 day weather forecasts</p>
      <Input
        aria-label="search-input"
        placeholder="Enter City Name here ..."
        type="text"
        value={query}
        onChange={($event) => handleInputChange($event.target.value)}
      />
    </div>
  );
}

SearchForm.propTypes = {
  onSearch: PropTypes.func.isRequired,
};

export default SearchForm;
