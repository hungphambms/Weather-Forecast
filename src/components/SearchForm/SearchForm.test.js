import { render, cleanup, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";

import SearchForm from "./index";

const INPUT_CHANGE_EVENT = {
  target: { value: "AAA" },
};

afterEach(cleanup);

describe("SearchForm component rendering", () => {
  it("renders SearchForm component without crashing", () => {
    const component = renderer.create(
      <SearchForm onSearch={(value) => console.log(value)} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("text in state is changed when input changed", () => {
    const { getByLabelText } = render(
      <SearchForm onSearch={(value) => console.log(value)} />
    );
    const inputContainer = getByLabelText("search-input");

    expect(inputContainer.value).toBe("");
    fireEvent.change(inputContainer, INPUT_CHANGE_EVENT);
    expect(inputContainer.value).toBe(INPUT_CHANGE_EVENT.target.value);
  });
});
