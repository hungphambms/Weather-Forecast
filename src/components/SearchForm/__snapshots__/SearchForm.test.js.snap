// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component rendering renders SearchForm component without crashing 1`] = `
<div
  className="searchContainer"
>
  <h1>
    Weather Forecast
  </h1>
  <p>
    Find the most current and reliable 5 day weather forecasts
  </p>
  <input
    aria-label="search-input"
    className="form-control"
    onChange={[Function]}
    placeholder="Enter City Name here ..."
    type="text"
    value=""
  />
</div>
`;
